src/coinsselectionalgorithm.cpp

```diff
@@ -0,0 +1,361 @@
+#include "coinsselectionalgorithm.hpp"
+#include <chrono>
+#include <cstdio>
+#include <iostream>
+#include <string>
+
+/* ---------- CCoinsSelectionAlgorithm ---------- */
+
+CCoinsSelectionAlgorithm::CCoinsSelectionAlgorithm(CoinsSelectionAlgorithmType _type,
+                                                   std::vector<std::pair<CAmount, size_t>> _amountsAndSizes,
+                                                   CAmount _targetAmount,
+                                                   CAmount _targetAmountPlusOffset,
+                                                   size_t _availableTotalSize)
+                                                   : type {_type},
+                                                     problemDimension{(int)_amountsAndSizes.size()},
+                                                     maxIndex{(int)_amountsAndSizes.size() - 1},
+                                                     amounts {PrepareAmounts(_amountsAndSizes)},
+                                                     sizes {PrepareSizes(_amountsAndSizes)},
+                                                     targetAmount {_targetAmount},
+                                                     targetAmountPlusOffset {_targetAmountPlusOffset},
+                                                     availableTotalSize {_availableTotalSize}
+{
+    tempSelection = new bool[problemDimension];    
+    optimalSelection = new bool[problemDimension];
+    for (int index = 0; index < problemDimension; ++index)
+    {
+        tempSelection[index] = false;
+        optimalSelection[index] = false;
+    }
```

Comment:
Since C++11 this loop for initializing the values of arrays is totally unnecessary.
Simply use:
```
    tempSelection = new bool[problemDimension]{};    
    optimalSelection = new bool[problemDimension]{};
```

Reply:
I agree with your suggestion.
Modified accordingly.